Content.Server/PowerCell/PowerCellSystem.cs

```diff
@@ -75,6 +76,12 @@ private void OnChargeChanged(EntityUid uid, PowerCellComponent component, ref Ch
             if (itemSlot.Item == uid)
                 RaiseLocalEvent(container.Owner, new PowerCellChangedEvent(false));
         }
+        // Mono edit - Also check if there are any batteries in mech
+        else if (_containerSystem.TryGetContainingContainer((uid, null, null), out var mechContainer)
+                 && mechContainer.Contains(uid))
+        {
+            RaiseLocalEvent(mechContainer.Owner, new PowerCellChangedEvent(false));
+        }
```

Comment:
don't need the upper check with this, can just make it fire if powercell is in any container
also are you sure this doesn't break IPCs holding batteries or backpacks that are batteries when you put a battery in it and such
if it does break make it use its own new event